api/oneclick.py

```diff
@@ -155,16 +170,157 @@ def circulate_item(self, patron_id, item_id, return_item=False):
             resp_dict = response.json()
             message = resp_dict.get('message', None)
 
-        error_response = self.validate_response(response=response, message=message, action=action)
+        self.validate_response(response=response, message=message, action=action)
 
         return resp_dict
 
 
+    def fulfill(self, patron, pin, licensepool, internal_format):
+        """ Get the actual resource file to the patron.
+        :return a FulfillmentInfo object.
+        """
+
+        patron_oneclick_id = self.validate_patron(patron)
+        (item_oneclick_id, item_media) = self.validate_item(licensepool)
+
+        # find patron's checkouts
+        checkouts_list = self.get_patron_checkouts(patron_id=patron_oneclick_id)
+        if not checkouts_list:
+            raise NoActiveLoan("Cannot fulfill %s - patron %s/%s has no checkouts.", item_oneclick_id, 
+                patron.authorization_identifier, patron_oneclick_id)
+
+        # find this licensepool in patron's checkouts
+        found_checkout = None
+        for checkout in checkouts_list:
+            if checkout.identifier == item_oneclick_id:
+                found_checkout = checkout
+                break
+        if not found_checkout:
+            raise NoActiveLoan("Cannot fulfill %s - patron %s/%s has no such checkout.", item_oneclick_id, 
+                patron.authorization_identifier, patron_oneclick_id)
+        
+        return found_checkout.fulfillment_info
+
+
+
+    def place_hold(self, patron, pin, licensepool, notification_email_address):
+        """Place a book on hold.
+
+        :param patron: a Patron object for the patron who wants to check out the book.
+        :param pin: The patron's password (not used).
+        :param licensepool: The Identifier of the book to be checked out is 
+        attached to this licensepool.
+        :param internal_format: Represents the patron's desired book format.  Ignored for now.
```

Comment:
This isn't relevant to this method, and if you copied it from some other place_hold implementation, the other docstring is probably also wrong.